components/central-application-gateway/pkg/authorization/oauth/oauthclient.go

```diff
@@ -58,6 +60,22 @@ func (c *client) GetToken(clientID, clientSecret, authURL string, headers, query
 	return tokenResponse.AccessToken, nil
 }
 
+func (c *client) GetTokenMTLS(clientID, authURL string, cert tls.Certificate, headers, queryParameters *map[string][]string) (string, apperrors.AppError) {
+	token, found := c.tokenCache.Get(clientID)
+	if found {
+		return token, nil
+	}
+
+	tokenResponse, err := c.requestTokenMTLS(clientID, authURL, cert, headers, queryParameters)
+	if err != nil {
+		return "", err
+	}
+
+	c.tokenCache.Add(clientID, tokenResponse.AccessToken, tokenResponse.ExpiresIn)
+
+	return tokenResponse.AccessToken, nil
+}
+
 func (c *client) InvalidateAndRetry(clientID, clientSecret, authURL string, headers, queryParameters *map[string][]string) (string, apperrors.AppError) {
 	c.tokenCache.Remove(clientID)
 
```

Comment:
This function assumes that token must be fetched with only `requestToken` function. MTLS Tokens will not be refreshed correctly